Build failure: retroarchFull because of libretro citra core

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *retroarchFull*

### Build log

```
error: builder for '/nix/store/5656wcj8bwi6ziyjjrc94bj3c07x53vp-libretro-citra-unstable-2024-01-24.drv' failed with exit code 2;
       last 10 log lines:
       > ./src/common/assert.h:33:15: note: in definition of macro 'ASSERT'
       >    33 |         if (!(_a_)) {                                                                              \
       >       |               ^~~
       > src/audio_core/hle/ffmpeg_decoder.cpp:215:56: error: 'struct AVFrame' has no member named 'channels'
       >   215 |                 response.num_channels = decoded_frame->channels;
       >       |                                                        ^~~~~~~~
       > src/audio_core/hle/ffmpeg_decoder.cpp:222:75: error: 'struct AVFrame' has no member named 'channels'
       >   222 |                     for (std::size_t channel(0); channel < decoded_frame->channels; channel++) {
       >       |                                                                           ^~~~~~~~
       > make: *** [Makefile:417: src/audio_core/hle/ffmpeg_decoder.o] Error 1
```

### Additional context

When adding retroarchFull to your system packages it will fail to build because of the citra core.
It seems like the citra core is incompatible with the latest ffmpeg module.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@aanderse
@edwtjo
@thiagokokada
@matthewbauer 
@kolbycrouch

### Workaround
Use ffmpeg_6 as build input: https://github.com/NixOS/nixpkgs/blob/9801748d78845bda7f7b9d854edccef173a04c35/pkgs/applications/emulators/retroarch/cores.nix#L321

### Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/emulators/retroarch/cores.nix
+++ b/pkgs/applications/emulators/retroarch/cores.nix
@@ -8,7 +8,7 @@
 , curl
 , fetchFromGitHub
 , fetchpatch
-, ffmpeg
+, ffmpeg_6
 , fluidsynth
 , fmt
 , freetype
@@ -318,7 +318,7 @@ in
 
   citra = mkLibretroCore rec {
     core = "citra";
-    extraBuildInputs = [ libGLU libGL boost ffmpeg nasm ];
+    extraBuildInputs = [ libGLU libGL boost ffmpeg_6 nasm ];
     makefile = "Makefile";
     makeFlags = [
       "HAVE_FFMPEG_STATIC=0"

